cabal.project

```diff
@@ -1,3 +1,8 @@
 packages: .
 tests: True
 benchmarks: True
+
+
+allow-newer:
+  , splitmix:base
+  , tasty-inspection-testing:base
```

Comment:
Let's keep `cabal.project` unchanged please: these `allow-newer` tend to fossilize and then no one remembers which version of GHC they are for. Besides, `tasty-inspection-testing` has been bumped already and I imagine `splitmix` will follow the suite soon.

Reply:
I have removed the `allowed-newer` for `tasty-inspection-testing` and bracketed it with `if impl (ghc >= 9.12)`